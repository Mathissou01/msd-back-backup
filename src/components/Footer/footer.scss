@use "src/styles/01-settings/settings.colors" as colors;

.c-Footer {
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-top: 1px solid colors.$gray-200;

  &__Version {
    margin-left: 1em;
    font-size: 16px;
  }

  &__ContentContainer {
    position: relative;
    z-index: 10;
    background-color: colors.$white;
  }

  &__Content {
    display: flex;
    justify-content: right;
    height: 72px;
  }

  &__LogoSuez {
    margin: auto 30px auto 0;
  }
}
