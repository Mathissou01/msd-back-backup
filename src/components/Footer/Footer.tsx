import "./footer.scss";
import Image from "next/image";

interface IFooterProps {
  isRoot?: boolean;
}

export default function Footer({ isRoot = false }: IFooterProps) {
  return (
    <>
      {!isRoot && (
        <footer className="c-Footer">
          <div className="c-Footer__Content">
            <Image
              className="c-Footer__LogoSuez"
              src="/images/suez-logo.svg"
              alt="logo_footer"
              width={110}
              height={50}
            />
          </div>
        </footer>
      )}
    </>
  );
}
