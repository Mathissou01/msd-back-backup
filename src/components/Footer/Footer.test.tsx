import { render } from "@testing-library/react";
import Footer from "./Footer";

describe("Footer", () => {
  it("renders a basic footer", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
