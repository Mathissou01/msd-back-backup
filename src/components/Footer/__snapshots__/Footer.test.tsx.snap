// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders a basic footer 1`] = `
<div>
  <div
    class="c-Footer"
  >
    <span
      class="c-Footer__Version"
    >
      Version - 
      1.1.0
    </span>
    <footer
      class="c-Footer__ContentContainer"
    >
      <div
        class="c-Footer__Content"
      >
        <img
          alt="logo_footer"
          class="c-Footer__LogoSuez"
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/images/suez-logo.svg"
          style="color: transparent;"
          width="110"
        />
      </div>
    </footer>
  </div>
</div>
`;
